promql/prettier/prettier_test.go

```diff
@@ -0,0 +1,966 @@
+		expected: `sum without(job, instance, foo) (metric_first{instance="first", foo="bar"} + metric_second{instance="second", foo="bar"})`,
+	},
+	// with comments
+	{
+		expr: `sum # comment
+(go_alloc_bytes) by (job)`,
+		expected: `sum by (job) # comment
+(go_alloc_bytes)`,
+	},
+	{
+		expr: `sum # comment
+(go_alloc_bytes) by # comment
+(job)`,
+		expected: `sum by # comment
+(job) # comment
+(go_alloc_bytes)`,
+	},
+	{
+		expr: `sum (go_alloc_bytes) # comment
+by (job)`,
+		expected: `sum by (job) (go_alloc_bytes) # comment`,
+	},
+	{
+		expr: `sum ( # comment
+go_alloc_bytes # comment
+) # comment
+by (job)`,
+		expected: `sum by (job) ( # comment
+go_alloc_bytes # comment
+) # comment`,
+	},
+	{
+		expr: `sum (go_alloc_bytes) # comment
+by ( # comment
+job # comment
+) # comment`,
+		expected: `sum by ( # comment
+job # comment
+) (go_alloc_bytes) # comment
+# comment`,
+	},
+	{
+		expr: `# comment
+{__name__="metric_name"}`,
+		expected: `# comment
+metric_name`,
+	},
+	{
+		expr: `{ # comment
+__name__="metric_name"}`,
+		expected: `metric_name{ # comment
+}`,
+	},
+	// We should not format those cases that contain comments within __name__ label-value.
+	{
+		expr: `{__name__ # comment
+="metric_name"}`,
+		expected: `{__name__ # comment
+="metric_name"}`,
+	},
+	{
+		expr: `{__name__= # comment
+"metric_name"}`,
+		expected: `{__name__= # comment
+"metric_name"}`,
+	},
+	{
+		expr: `{__name__="metric_name" # comment
+}`,
+		expected: `metric_name{ # comment
+}`,
+	},
+	{
+		expr:     `{__name__="metric_name"} # comment`,
+		expected: `metric_name # comment`,
+	},
+}
+
+func TestLexItemSorting(t *testing.T) {
+	prettier := New("")
+	for i, expr := range sortingLexItemCases {
+		expectedSlice := prettier.refreshLexItems(prettier.lexItems(expr.expected))
+		input := prettier.lexItems(expr.expr)
+		testutil.Equals(t, expectedSlice, prettier.sortItems(input), "%d: input %q", i, expr.expr)
+	}
+}
+
+var prettierCases = []prettierTest{
+	{
+		expr:     `go_goroutines`,
+		expected: `  go_goroutines`,
+	},
+	{
+		expr:     `go_goroutines{job="prometheus", instance="localhost:9090"}`,
+		expected: `  go_goroutines{job="prometheus", instance="localhost:9090"}`,
+	},
+	{
+		expr:     `go_goroutines{job="prometheus",instance="localhost:9090"}`,
+		expected: `  go_goroutines{job="prometheus", instance="localhost:9090"}`,
+	},
+	{
+		expr: `instance_cpu_time_ns{app="lion", proc="web", rev="34d0f99", env="prod", job="cluster-manager", host="localhost"}`,
+		expected: `  instance_cpu_time_ns{
+    app="lion",
+    proc="web",
+    rev="34d0f99",
+    env="prod",
+    job="cluster-manager",
+    host="localhost",
+  }`,
+	},
+	{
+		expr: `instance_cpu_time_ns{app="lion", proc="web", rev="34d0f99", env="prod", job="cluster-manager", host="localhost",}`,
+		expected: `  instance_cpu_time_ns{
+    app="lion",
```

Comment:
This is more vertical than it needs to be, wrap so there's more than one matcher per line.

Reply:
Do you mean, for those vector_selectors that violate column limit, keep their label_matchers on a different line only? 
Example: for the current scenario,

```
instance_cpu_time_ns{
  app="lion", proc="web", rev="34d0f99", env="prod", job="cluster-manager", host="localhost",
}
```

Reply:
yeah, avoid having one matcher per line anyway

Reply:
Addressed. Now, all label marchers will be in a single newline (like the mentioned example in the above comment), rather than putting them on new lines each time.